action/protocol/staking/protocol.go

```diff
@@ -331,21 +354,25 @@ func (p *Protocol) ReadState(ctx context.Context, sr protocol.StateReader, metho
 		return nil, errors.Wrap(err, "failed to unmarshal request")
 	}
 
-	center, err := getOrCreateCandCenter(sr)
```

Comment:
this is not necessary for some cases, such as readStateBuckets and readStateBucketsByVoter. From which we can tell that it is not necessary to create candidate center here (which may be time consuming).